src/NzbDrone.Core/Indexers/TorrentRssIndexer/TorrentRssParserFactory.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using NLog;
+
+namespace NzbDrone.Core.Indexers.TorrentRssIndexer
+{
+    public class TorrentRssParserFactory : ITorrentRssParserFactory
+    {
+        private static readonly ITorrentRssParserSettingsCache SettingsCache = new TorrentRssParserSettingsCache();
```

Comment:
No static. Use ```ICacheManager``` to get a nice persistent cache.
You also won't need to write your own threadsafe cache, CacheManager already does it. We use it throughout sonarr.